### ISSUE TITLE
Time of day change to fade in/out. (GNA #5915)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1066]
**[Original submission](https://gna.org/bugs/index.php?5915)** by irrevenant on 2006-05-07

The transition to and from night-time is a bit abrupt. could it take perhaps 1/2 a second fading in/out from the previous state between turns?

(Reproduced on Win98SE)
Release: 1.1.2a
Priority: 5 - Normal
Severity: 1 - Wish

**Comment by Vultraz** [comment]
Still not implemented 12 years later. Honestly, best to wait on this until 1.15 and accelerated rendering.

**Comment by sevu** [comment]
WIN 98 SE !!!

**Comment by cooljeanius** [comment]
> @Vultraz is this something that could be implemented now?

You mean with @mesilliac's enhancements to the graphics subsystems? (Adding hardware acceleration and all that...)

**Comment by Pentarctagon** [comment]
Yep.

**Comment by mesilliac** [comment]
IIRC the ToD stuff wasn't accelerated as it was waiting on a newer SDL function. Probably SDL_RenderGeometry.

It would be easier if it were a blanket effect, but it's blended in a complex way between sextants of a hex. For example in caves there is no ToD effect, so the colour modifier has to blend smoothly between inside and outside areas.

**Comment by Pentarctagon** [comment]
Any chance you'd be able to take a look at any of #7657 btw?

**Comment by mesilliac** [comment]
I commented on a couple, but the rest i could either never reproduce (like scaling artifacts and scrolled stuff not drawing) or they deal with systems i never went into (like animations).

7617, 7615, 6922 should be easy to fix but i don't really have the time or inclination.

**Comment by Pentarctagon** [comment]
@mesilliac side question: is it documented anywhere which things aren't accelerated?

**Comment by mesilliac** [comment]
I don't think so. The only things that jump to mind are time-of-day lighting, UI blur, and the submerge effect.

IPF operations are done in software, but it's not really reasonable to do those in hardware anyway.
